xbmc/filesystem/File.cpp

```diff
@@ -536,6 +530,17 @@ ssize_t CFile::Read(void *lpBuf, size_t uiBufSize)
 
   try
   {
+    if (uiBufSize == 0)
+    {
+      // "test" read with zero size
+      // some VFSs don't handle correctly null buffer pointer
+      // provide valid buffer pointer for them
+      const int chunkSize = m_pFile->GetChunkSize();
+      auto_buffer dummyBuf(chunkSize > 1 ? (size_t)chunkSize : 1024);
```

Comment:
Uh, why 1024 as a fallback? Seems like a random number (without a comment as to why).